app/full_text_search.py

```diff
@@ -114,47 +98,74 @@ def create_romanized_query_txt(query_txt):
     return enclose_in_parenthesis(romanized_query)
 
 
-def get_person_ids_from_results(romanized_query, results_list):
+def is_query_match(query_txt, values, romanized_values):
+    """
+    To check if the query match the field value
+    Args:
+        query_txt: Search query
+        values: field value for unidecode method
+        romanized_values: romanized field value
+    Returns:
+        Boolean
+    """
+    if not query_txt:
+        return True
+
+    romanized_query_list = (script_variant.romanize_search_query(query_txt))
+
+    for name in romanized_query_list:
+        words = name.split(" ")
+        for word_index, word in enumerate(words):
+            if not re.search(word, values + " " + romanized_values, re.I):
+                break
+            if word_index == len(words) - 1:
+                return True
+    return False
```

Comment:
This is for creating compiled regular expression for AND search. We want to make sure that every part of the name and location query match the name and location field. Unfortunately, there is no such symbol: "&" in the re.search so we combine the string and see if the query contain part of the field. Here is an example: 
If the user input the location query: tokyo
Inside the field there have: tokyo japan
And this is valid search and also you can get the result.
However, if user input the query: "tokyo taiwan"
This this not valid and will return None because it doesn't match the location field.
![screenshot from 2016-08-30 15 02 55](https://cloud.githubusercontent.com/assets/15961861/18077839/131fd7d6-6ec3-11e6-90c3-2ce0c7953030.png)

For the old version of full text search, we use OR search because it only has one input box and we don't know which part of the query match the name field. (and in this cause if you input tokyo taiwan as location query it will work because the tokyo part match the location field and this will be a little bit weird and if we use the OR, the user can type random things into the search boxs and still get the result) ---> the goal is to make the location query more accurate to higher the accuracy of finding the right person.